### ISSUE TITLE
Using Waffle 3 without ganache-core

### ISSUE THREAD

**Issue by alcuadrado** [EthWorks/Waffle#254]
Hey guys,

I'm opening this issue because I've been having multiple issues with `ganache-core`'s dependencies lately. For example, it installs husky in the user's repository, which makes no sense. It also introduces lots of unnecessary dependencies and multiple `npm audit` warnings.

We don't actually use it on Buidler, we just need `@waffle/provider` because of its type definitions. We'd also like to be able to use `loadFixture`, but it only works with your `ganache-core`-based `MockProvider`.

Can v3 be refactored in a way that getting those things doesn't require installing `ganache-core`?

Thanks!

**Comment by sz-piotr** [comment]
We'd love to be able to not use ganache-core. We feel the bloat every day. In fact it was one of the ideas behind https://github.com/ethereum-ts/deth. 

The biggest obstacle to not using ganache is: what else should we use?
1) We can't use nothing, because we need to have an in-memory blockchain for everything
2) We can't use buidler (we tried) because it is not a library - it assumes a lot of thing about the environment and I think importing from buidler causes side-effects (or using buidlers evm, can't remember) https://github.com/EthWorks/Waffle/issues/254
3) We can't use deth, because we do not have the resources to maintain it, and it is missing a lot of features

**Comment by alcuadrado** [comment]
I agree that using `ganache-core` for the default installation of Waffle is the way to go, at least for now. 

What I'd like to able able to do is to use some/most of the internal `@waffle/` packages without them pulling `ganache-core`.

I think this should be doable with this changes:
1. Define the provider interface in a package that doesn't use `ganache-core`.
2. Have a separate package with your `ganache-core`-based `MockProvider`.
3. Make sure that no other `@waffle/*` package depends on the `MockProvider`, but on the generic provider interface. All the functions that use a provider should get one as a required param.
4. Wrap all those functions in `ethereum-waffle` to use `MockProvider` as the default provider. 

What do you think?
